gymnasium/experimental/wrappers/lambda_observations.py

```diff
@@ -0,0 +1,327 @@
+        >>> import gymnasium as gym
+        >>> env = gym.make("CarRacing-v1")
+        >>> env.observation_space.shape
+        (96, 96, 3)
+        >>> grayscale_env = GrayscaleObservationV0(env)
+        >>> grayscale_env.observation_space.shape
+        (96, 96)
+        >>> grayscale_env = GrayscaleObservationV0(env, keep_dim=True)
+        >>> grayscale_env.observation_space.shape
+        (96, 96, 1)
+    """
+
+    def __init__(self, env: gym.Env, keep_dim: bool = False):
+        """Constructor for an RGB image based environments to make the image grayscale."""
+        assert isinstance(env.observation_space, spaces.Box)
+        assert (
+            len(env.observation_space.shape) == 3
+            and env.observation_space.shape[-1] == 3
+        )
+        assert (
+            np.all(env.observation_space.low == 0)
+            and np.all(env.observation_space.high == 255)
+            and env.observation_space.dtype == np.uint8
+        )
+
+        self.keep_dim = keep_dim
+        if keep_dim:
+            new_observation_space = spaces.Box(
+                low=0,
+                high=255,
+                shape=env.observation_space.shape[:2] + (1,),
+                dtype=np.uint8,
+            )
+            super().__init__(
+                env,
+                lambda obs: jp.expand_dims(
+                    jp.sum(
+                        jp.multiply(obs, jp.array([0.2125, 0.7154, 0.0721])), axis=-1
+                    )
+                ),
+                new_observation_space,
+            )
+        else:
+            new_observation_space = spaces.Box(
+                low=0, high=255, shape=env.observation_space.shape[:2], dtype=np.uint8
+            )
+            super().__init__(
+                env,
+                lambda obs: jp.sum(
+                    jp.multiply(obs, jp.array([0.2125, 0.7154, 0.0721])), axis=-1
+                ),
+                new_observation_space,
+            )
+
+
+class ResizeObservationV0(LambdaObservationV0):
+    """Observation wrapper for resize image observations using opencv.
+
+    Example:
+        >>> import gymnasium as gym
+        >>> env = gym.make("CarRacing-v1")
+        >>> resized_env = ResizeObservationV0(env, (32, 32))
+        >>> resized_env.observation_space.shape
+        (32, 32, 3)
+    """
+
+    def __init__(self, env: gym.Env, shape: tuple[int, ...]):
+        """Constructor that requires an image environment observation space with a shape."""
+        assert isinstance(env.observation_space, spaces.Box)
+        assert len(env.observation_space.shape) in [2, 3]
+        assert np.all(env.observation_space.low == 0) and np.all(
+            env.observation_space.high == 255
+        )
+        assert env.observation_space.dtype == np.uint8
+
+        assert isinstance(shape, tuple)
+        assert all(np.issubdtype(type(elem), np.integer) for elem in shape)
+        assert all(x > 0 for x in shape)
+
+        self.shape = tuple(shape)
+
```

Comment:
I'll move this after the `try...except` block to keep all the validation separated from logic and variable assignment; also the cv2 check I think should be the first check to do (just for style to avoid having floating import merged in code logic)

Maybe, but this is more a matter of preferences, since the validation block is a bit long I'll consider also moving it out as a separated method to keep the `__init__` more concise.

This may be valid also for other wrappers which has a long validation before actually doing stuff

Reply:
Done, I will check other wrappers